allauth_2fa/views.py

```diff
@@ -66,20 +66,16 @@ class TwoFactorSetup(FormView):
     success_url = reverse_lazy('two-factor-backup-tokens')
 
     def dispatch(self, request, *args, **kwargs):
+        if request.user.totpdevice_set.filter(confirmed=True).exists():
+            return HttpResponseRedirect(reverse_lazy('two-factor-backup-tokens'))
 
-        if 'allauth_otp_qr_secret_key' not in request.session:
-            self.secret_key = random_hex(20).decode('ascii')
-            request.session['allauth_otp_qr_secret_key'] = self.secret_key
-        else:
-            self.secret_key = request.session['allauth_otp_qr_secret_key']
+        request.user.totpdevice_set.filter(confirmed=False).delete()
+        TOTPDevice(user=request.user, confirmed=False).save()
```

Comment:
This change doesn't seem to actually work -- when attempting to verify a device, dispatch runs *before* the form checking, thus you actually end up deleting the device you're trying to verify right before verifying it!